### PR TITLE
[JENKINS-68100] Support updating @Deprecated and @RestrictedSince TODOS

### PR DESCRIPTION
**Pull request by kaltepeter**
See [JENKINS-68100](https://issues.jenkins.io/browse/JENKINS-68100).

No tests are included in the pull request because it is updating a script the maintainers run and it would be kind of difficult to test since it depends on the state of git.

### Testing done

To just test the git grep part I ran: `git grep --line-number -E '@Deprecated\(since = "TODO"\)|@since TODO|@RestrictedSince\("TODO"\)' -- '*.java' '*.jelly' '*.js'` on my local mac. 

I was able to test `@Deprecated(since = "TODO")` and `@since TODO` as they exist already. To test `@RestrictedSince("TODO")` I could only test a non-real local commit, I changed a version back to 'TODO' in a file that was already replaced and committed to local temporarily and ran the script.

Attaching a video of the test and output since there is a lot going on.

https://user-images.githubusercontent.com/5103752/194767008-0f09db7e-594d-4533-ade0-dfca1ab9b05f.mov

And screen from gitpod run(linux):
image

### Proposed changelog entries

- Adds `@Deprecated(since = "TODO")` and `@RestrictedSince("TODO")` to the find and replace in the [since generator](https://github.com/jenkinsci/jenkins/blob/master/update-since-todo.py)

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7238"></a>

### CODE REVIEW

File: update-since-todo.py
Comment (kaltepeter) [inline update-since-todo.py:61]: This was manually tested as providing the same functionality, the latter patterns would have gotten unruly so I applied the same pattern just for consistency.

**Review by NotMyFault — APPROVED** [review]
Works and looks fine, thanks for taking care of the issue!

File: update-since-todo.py
Comment (basil) [inline update-since-todo.py:61]: Prefer f-strings for consistency with the convention used elsewhere in this file:

```suggestion
                f"@since {since_version}",
```

File: update-since-todo.py
Comment (basil) [inline update-since-todo.py:67]: Prefer f-strings (and a comma after the last element) for consistency with the convention used elsewhere in this file:

```suggestion
                f'@Deprecated(since = "{since_version}")',
```

File: update-since-todo.py
Comment (basil) [inline update-since-todo.py:73]: Prefer f-strings for consistency with the convention used elsewhere in this file:

```suggestion
                f'@RestrictedSince("{since_version}")',
```

**Review by basil — APPROVED** [review]
Thanks for the PR!

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
